extensions/iceberg/src/main/java/io/deephaven/iceberg/util/SortOrderProvider.java

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+/**
+ * A specification for providing {@link org.apache.iceberg.SortOrder} while writing to an iceberg table.
+ */
+public interface SortOrderProvider {
+
+    // Static factory methods for creating SortOrderProvider instances
+
+    /**
+     * Do not sort the data while writing new data to the iceberg table.
+     */
+    static SortOrderProvider disableSorting() {
```

Comment:
Not a fan of duplication in naming; I would just call this `disabled` or `unsorted`.